lib/miq_expression.rb

```diff
@@ -3,352 +3,14 @@ class MiqExpression
-    'ContainerRoute'                              => 'container_route',
-    'ContainerService'                            => 'container_service',
-    'ContainerTemplate'                           => 'container_template',
-    'PersistentVolume'                            => 'persistent_volume'
-  }
-  EXCLUDE_FROM_RELATS = {
-    "ManageIQ::Providers::CloudManager" => ["hosts", "ems_clusters", "resource_pools"]
-  }
-
-  FORMAT_SUB_TYPES = {
-    :boolean     => {
-      :short_name => _("Boolean"),
-      :title      => _("Enter true or false")
-    },
-    :bytes       => {
-      :short_name => _("Bytes"),
-      :title      => _("Enter the number of Bytes"),
-      :units      => [
-        [_("Bytes"), :bytes],
-        [_("KB"), :kilobytes],
-        [_("MB"), :megabytes],
-        [_("GB"), :gigabytes],
-        [_("TB"), :terabytes]
-      ]
-    },
-    :date        => {
-      :short_name => _("Date"),
-      :title      => _("Click to Choose a Date")
-    },
-    :datetime    => {
-      :short_name => _("Date / Time"),
-      :title      => _("Click to Choose a Date / Time")
-    },
-    :float       => {
-      :short_name => _("Number"),
-      :title      => _("Enter a Number (like 12.56)")
-    },
-    :gigabytes   => {
-      :short_name => _("Gigabytes"),
-      :title      => _("Enter the number of Gigabytes")
-    },
-    :integer     => {
-      :short_name => _("Integer"),
-      :title      => _("Enter an Integer")
-    },
-    :kbps        => {
-      :short_name => _("KBps"),
-      :title      => _("Enter the Kilobytes per second")
-    },
-    :kilobytes   => {
-      :short_name => _("Kilobytes"),
-      :title      => _("Enter the number of Kilobytes")
-    },
-    :megabytes   => {
-      :short_name => _("Megabytes"),
-      :title      => _("Enter the number of Megabytes")
-    },
-    :mhz         => {
-      :short_name => _("Mhz"),
-      :title      => _("Enter the number of Megahertz")
-    },
-    :numeric_set => {
-      :short_name => _("Number List"),
-      :title      => _("Enter a list of numbers separated by commas")
-    },
-    :percent     => {
-      :short_name => _("Percent"),
-      :title      => _("Enter a Percent (like 12.5)"),
-    },
-    :regex       => {
-      :short_name => _("Regular Expression"),
-      :title      => _("Enter a Regular Expression")
-    },
-    :string      => {
-      :short_name => _("Text String"),
-      :title      => _("Enter a Text String")
-    },
-    :string_set  => {
-      :short_name => _("String List"),
-      :title      => _("Enter a list of text strings separated by commas")
-    }
-  }
-  FORMAT_SUB_TYPES[:fixnum] = FORMAT_SUB_TYPES[:decimal] = FORMAT_SUB_TYPES[:integer]
-  FORMAT_SUB_TYPES[:mhz_avg] = FORMAT_SUB_TYPES[:mhz]
-  FORMAT_SUB_TYPES[:text] = FORMAT_SUB_TYPES[:string]
+  CONFIG = YAML.load(ERB.new(File.read(Rails.root.join("config", "miq_expression.yml"))).result)
```

Comment:
I **think** that if you make the `config` variable lower case, it will get garbage collected after the class load -> lower memory footprint overall.